packages/contentstack-export/src/lib/export/assets.js

```diff
@@ -75,212 +79,223 @@ ExportAssets.prototype = {
+            return resolve()
           }
+
           return self
             .getFolderJSON(0, fCount)
             .then(function () {
               // asset folders have been successfully exported
-              addlogs(config, 'Asset-folders have been successfully exported!', 'success');
+              addlogs(self.config, 'Asset-folders have been successfully exported!', 'success');
               return resolve();
-            })
-            .catch(function (error) {
-              addlogs(config, chalk.red('Error while exporting asset-folders!'), 'error');
+            }).catch(function (error) {
+              addlogs(self.config, chalk.red('Error while exporting asset-folders!'), 'error');
               return reject(error);
-            });
+            })
         })
         .catch(function (error) {
-          addlogs(config, error, 'error');
+          addlogs(self.config, error, 'error')
           // error while fetching asset folder count
-          return reject(error);
+          return reject(error)
         });
     });
-  },
-  getFolderJSON: function (skip, fCount) {
-    let self = this;
+  }
+
+  getFolderJSON(skip, fCount) {
+    let self = this
     return new Promise(function (resolve, reject) {
       if (typeof skip !== 'number') {
-        skip = 0;
+        skip = 0
       }
+
       if (skip >= fCount) {
-        helper.writeFile(folderJSONPath, self.folderData);
-        return resolve();
+        helper.writeFile(self.folderJSONPath, self.folderData)
+        return resolve()
       }
 
       const queryRequestObj = {
+        skip,
         include_folders: true,
-        query: { is_dir: true },
-        skip: skip,
-      };
+        query: { is_dir: true }
+      }
 
-      client
-        .stack({ api_key: config.source_stack, management_token: config.management_token })
+      self.client
+        .stack({ api_key: self.config.source_stack, management_token: self.config.management_token })
         .asset()
         .query(queryRequestObj)
         .find()
         .then((response) => {
-          response.items.forEach(function (folder) {
-            self.folderData.push(folder);
-          });
-          skip += 100;
-          return self.getFolderJSON(skip, fCount).then(resolve).catch(reject);
-        });
-    });
-  },
-  getAssetCount: function (folder) {
+          skip += 100
+          self.folderData.push(...response.items)
+          return self.getFolderJSON(skip, fCount).then(resolve).catch(reject)
+        })
+    })
+  }
+
+  getAssetCount(folder) {
+    const self = this
     return new Promise(function (resolve, reject) {
       if (folder && typeof folder === 'boolean') {
-        let queryOptions = { include_folders: true, query: { is_dir: true }, include_count: true };
-        client
-          .stack({ api_key: config.source_stack, management_token: config.management_token })
+        let queryOptions = {
+          skip: 99999990,
```

Comment:
Why skip is 99999990 ?

Reply:
This method is meant to get only counts so we doing max/large number of skip to skip fetching content and trying to get only count